modules/build/src/main/scala/scala/build/bsp/BspServer.scala

```diff
@@ -20,11 +23,29 @@ class BspServer(
     with ScalaDebugServerForwardStubs
     with ScalaBuildServerForwardStubs with JavaBuildServerForwardStubs with HasGeneratedSources {
 
+  private var client: Option[BuildClient] = None
+
+  override def onConnectWithClient(client: BuildClient): Unit = this.client = Some(client)
+
   private var extraDependencySources: Seq[os.Path] = Nil
   def setExtraDependencySources(sourceJars: Seq[os.Path]): Unit = {
     extraDependencySources = sourceJars
   }
 
+  // Can we accept some errors in some circumstances?
+  override protected def onFatalError(throwable: Throwable): Unit = {
+    val sw = new StringWriter()
+    throwable.printStackTrace(new PrintWriter(sw))
```

Comment:
I think this discards the cause exceptions, which are often helpful. Better loop on the causes, like done [here](https://github.com/VirtusLab/scala-cli/blob/5d714dceaf4aa8049a4cc778725f8c47f7e3403f/modules/build/src/main/scala/scala/build/Build.scala#L809-L816).

Reply:
Are you sure? I run some simple test and it prints nicely nested exceptions (I am using `printStackTrace` like 10 years now :)

```
⋊> ~/w/P/scala-cli-pg bat errors.scala                                                                                                         10:50:36
   1   │ @main def errors = 
   2   │  try
   3   │    try ???
   4   │    catch 
   5   │      case e: Throwable => throw RuntimeException(e)
   6   │  catch
   7   │    case e => e.printStackTrace()
⋊> ~/w/P/scala-cli-pg scala-cli errors.scala                                                                                                   10:50:40
java.lang.RuntimeException: scala.NotImplementedError: an implementation is missing
	at errors$package$.errors(errors.scala:5)
	at errors.main(errors.scala:1)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at scala.cli.runner.Runner$.main(Runner.scala:14)
	at scala.cli.runner.Runner.main(Runner.scala)
Caused by: scala.NotImplementedError: an implementation is missing
	at scala.Predef$.$qmark$qmark$qmark(Predef.scala:344)
	at errors$package$.errors(errors.scala:3)
	... 7 more

```

Reply:
Oh, I didn't know that, seems that method is more evolved than it looks!